classes/user/form/ContactForm.php

```diff
@@ -114,6 +114,7 @@ public function readInputData()
     public function execute(...$functionArgs)
     {
         $user = $this->getUser();
+        $functionArgs['emailUpdated'] = $user->getEmail() !== $this->getData('email');
```

Comment:
@asmecher  The reason we are not updating the session and clearing out all but existing session here is because the data commit is not happening here but in the `PKP\user\form\BaseProfileForm::execute` and we don't make those changes until the changes are pushed to DB itself . Thats why we are setting just a flag here .